mantidproject/mantid
# Enable SymmetryOperation to have vector-components >= 1

Fixes #14970.

To enable the requested behavior it seemed appropriate to do some minor cleanup in `SymmetryOperation` first, the creation of actual `std::pair<IntMatrix, V3R>` instances was dropped for some intermediate steps during parsing/creating in favor of `MatrixVectorPair<int, V3R>`.

The procedures for wrapping vector components to the appropriate interval are now part of `Group` where they are explicitly carried out on the symmetry operations.

Nevertheless, all of this is more or less internal and I don't think this needs to go into the release notes.

**To test:**

Review the code changes. Test as many cases of symmetry operations as you can think of and try to break everything. Here are some examples that work now but did not work before:

```
op1 = SymmetryOperationFactory.createSymOp("x+5,y+2,z-10/20")
print op1.transformCoordinates([0, 0, 0]) # should print [5,2,-0.5]

op2 = SymmetryOperationFactory.createSymOp("-z+1,-x-2,-y-1/2")
print op2.transformCoordinates([3, 2, 1]) # should print [0, -5, -2.5]

op3 = SymmetryOperationFactory.createSymOp("z+1+4-1/5, x, y")
print op3.transformCoordinates([3, 2, 1]) # should print [5.8, 3, 2]
```

Group now has to make sure that the symmetry operations used for construction are properly confined. Try some variations of this example and see if you can break it:

```
g = Group("x+5,y+2,z-10; x,y,z; -x,-y,-z")

print g.isGroup() # should print True
print g.getOrder() # should print 2
print g.getSymmetryOperationStrings() # should only print x,y,z and -x,-y,-z
```



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
